### ISSUE TITLE
[BUG] Very slow to submit forms with nested EditGrids and conditional visibility

### ISSUE THREAD

**Issue by peterhall-agrimetrics** [formio/formio#1251]
## Environment
Please provide as many details as you can:

- Hosting type
  - [] Form.io
  - [x] Local deployment
    - Version: 

## Steps to Reproduce
1. Create a form with one hidden component and an EditGrid
2. Within the EditGrid, create multiple EditGrids (eg 5) and give them conditional visibility based on the hidden component
3. Post a form submission with multiple (eg 6) rows at the top level, and a small amount of data in the inner EditGrids (eg 1 row for 1 of the EditGrids, the other 4 empty)

## Expected behavior
Submission processed in a reasonable amount of time

## Observed behavior
Submission took 18 seconds to be processed.
Without the conditional visibility, takes <4 seconds
Without the conditional visibility or the nesting of EditGrids (ie all EditGrids moved to the top level), a similarly-sized payload takes <0.5 seconds

**Comment by peterhall-agrimetrics** [comment]
As an additional note, the example I have given above is a simplified one, for the sake of reproducibility and client confidentiality. The example we are encountering in our real-world application is far more severe - the top level EditGrid also contains additional components (approx 20 to 30). A user tried to post a submission with 8 records in the outer EditGrid (and, as above, 4 of the inner EditGrids empty) and it took 5 minutes.

**Comment by peterhall-agrimetrics** [comment]
I have looked into this further and found that the performance issue is not directly related to the inner EditGrids, but appears when using "Advanced" conditional instead of "Simple" within an EditGrid. I have attached form specs files for a form that we created for testing. One uses simple conditional logic for the inner components, the other uses advanced, but the test they are performing are identical. The EditGrid just contains 7 normal Number components.

For the "Simple" case, a form submission with 3 records took 290ms. For the "Advanced" case, it took over 4 seconds.

[performance.json.advanced.txt](https://github.com/formio/formio/files/6478478/performance.json.advanced.txt)
[performance.json.simple.txt](https://github.com/formio/formio/files/6478479/performance.json.simple.txt)

**Comment by peterhall-agrimetrics** [comment]
We have discovered a workaround, which is to not use advanced conditional logic, and instead create a new hidden component for storing the value of a calculation, which can be used as the subject of a simple conditional logic test. This seems to run without any performance problem on the backend.

**Comment by abhishek0196** [comment]
Hello @peterhall-agrimetrics, can you please share the solution here, it would help me with one similar scenario.

**Comment by peterhall-agrimetrics** [comment]
> Hello @peterhall-agrimetrics, can you please share the solution here, it would help me with one similar scenario.

The solution is as outlined in my previous message. Have you tried that?

**Comment by abhishek0196** [comment]
> > Hello @peterhall-agrimetrics, can you please share the solution here, it would help me with one similar scenario.
> 
> The solution is as outlined in my previous message. Have you tried that?

No, I have to try it. I Was just wondering if you have anything handy which I can refer to.

**Comment by lane-formio** [comment]
Thanks for the contribution @peterhall-agrimetrics
